test/utility.h

```diff
@@ -91,16 +91,34 @@ class TestIntegration : public WasmVmIntegration {
 class TestContext : public ContextBase {
 public:
   TestContext(WasmBase *wasm) : ContextBase(wasm) {}
+  TestContext(WasmBase *wasm, const std::shared_ptr<PluginBase> &plugin)
+      : ContextBase(wasm, plugin) {}
```

Comment:
It almost feels like we should have a separate `TestVmContext` and `TestRootContext` classes.

I'm going to allow this for now, beause all Context types are combined in the production code, and existing tests operate on VmContext, which doesn't match the production code or SDKs...

Something to keep in mind in the future, @mpwarres.